src/graph/LookupExecutor.cpp

```diff
@@ -0,0 +1,707 @@
+            auto* right = rExpr->right();
+            /**
+             *  TODO (sky) : Does not support left expr and right expr are both kAliasProp.
+             *               Handle error (kAliasProp OP kAliasProp)
+             *               No handle error for (kAliasProp OP kFunctionCall(kAliasProp)) yet.
+             */
+            if (left->kind() == nebula::Expression::kAliasProp &&
+                right->kind() == nebula::Expression::kAliasProp) {
+                return Status::SyntaxError("Syntax error ：%s", rExpr->toString().c_str());
+            }  else if (left->kind() == nebula::Expression::kAliasProp) {
+                auto* aExpr = dynamic_cast<const AliasPropertyExpression*>(left);
+                prop = *aExpr->prop();
+                filters_.emplace_back(std::make_pair(prop, rExpr->op()));
+            } else if (right->kind() == nebula::Expression::kAliasProp) {
+                auto* aExpr = dynamic_cast<const AliasPropertyExpression*>(right);
+                prop = *aExpr->prop();
+                filters_.emplace_back(std::make_pair(prop, rExpr->op()));
+            } else {
+                return Status::SyntaxError("Unsupported expression ：%s",
+                                           rExpr->toString().c_str());
+            }
+            break;
+        }
+        case nebula::Expression::kFunctionCall : {
+            auto* fExpr = dynamic_cast<const FunctionCallExpression*>(expr);
+            auto* name = fExpr->name();
+            if (*name == "udf_is_in") {
+                return Status::SyntaxError("Unsupported function ： %s", name->c_str());
+            }
+            break;
+        }
+        default : {
+            return Status::SyntaxError("Syntax error ： %s", expr->toString().c_str());
+        }
+    }
+    return Status::OK();
+}
+
+Status LookupExecutor::checkFilter() {
+    auto status = traversalExpr(sentence_->whereClause()->filter());
+    if (!status.ok()) {
+        return status;
+    }
+    if (filters_.empty()) {
+        return Status::SyntaxError("Where clause error . have not index matching");
+    }
+    return Status::OK();
+}
+
+Status
+LookupExecutor::findValidIndex() {
+    std::vector<std::shared_ptr<nebula::cpp2::IndexItem>> indexes;
+    std::set<std::string> filterCols;
+    for (auto& filter : filters_) {
+        filterCols.insert(filter.first);
+    }
+    /**
+     * step 1 : found out all valid indexes. for example :
+     * tag (col1 , col2, col3)
+     * index1 on tag (col1, col2)
+     * index2 on tag (col2, col1)
+     * index3 on tag (col3)
+     *
+     * where where clause is below :
+     * col1 > 1 and col2 > 1 --> index1 and index2 are valid.
+     * col1 > 1 --> index1 is valid.
+     * col2 > 1 --> index2 is valid.
+     * col3 > 1 --> index3 is valid.
+     */
+    for (auto& index : indexes_) {
+        bool matching = true;
+        size_t filterNum = 1;
+        for (const auto& field : index->get_fields()) {
```

Comment:
I think the logic still has some problems. 
As the examples in your comments,  for "col1 > 1",  index1 should be valid. 
But as you logic here, index1 is invalid.  Right?

Reply:
Emm.. index1 really valid when col1 > 1,  and by this example only index1 is valid. 
here I using index's columns to check the index validity. Check field matching in order of columns, the loop will break when not matching column found.

Reply:
make sense